linera-execution/src/wasm/system_api.rs

```diff
@@ -330,6 +360,32 @@ macro_rules! impl_view_system_api_for_contract {
                 error.into()
             }
 
+            fn contains_key_new(&mut self, key: &[u8]) -> Result<Self::ContainsKey, Self::Error> {
+                BaseRuntime::contains_key_new(self, key.to_vec())
+            }
+
+            fn contains_key_wait(
+                &mut self,
+                promise: &Self::ContainsKey,
+            ) -> Result<bool, Self::Error> {
+                BaseRuntime::contains_key_wait(self, promise)
+            }
+
+            fn read_multi_values_bytes_new(
+                &mut self,
+                keys: Vec<&[u8]>,
+            ) -> Result<Self::ReadMultiValuesBytes, Self::Error> {
+                let keys = keys.iter().map(|x| x.to_vec()).collect::<Vec<_>>();
```

Comment:
nit: This should also work: `keys.iter().map(Vec::from).collect()`

Reply:
Actually, it does not work.

Reply:
Oh I see, you still need `collect::<Vec<_>>()`. Never mind then

Reply:
I think it needs to take ownership:

```suggestion
                let keys = keys.into_iter().map(Vec::from).collect();
```

Reply:
Thanks, that one works.